src/uu/ls/src/ls.rs

```diff
@@ -1106,7 +1106,17 @@ fn sort_entries(entries: &mut Vec<PathData>, config: &Config) {
             entries.sort_by_key(|k| Reverse(k.md.as_ref().map(|md| md.len()).unwrap_or(0)))
         }
         // The default sort in GNU ls is case insensitive
-        Sort::Name => entries.sort_by_key(|k| k.file_name.to_lowercase()),
+        Sort::Name => entries.sort_by_key(|k| {
+            let has_dot: bool = k.file_name.starts_with('.');
+            let filename_nodot: String = if has_dot {
```

Comment:
Names are usually short so this maybe a nitpick, but we could work with something like this that does not involve clone / collect:
```
let filename_nodot = &k.file_name[if has_dot { 1 } else { 0 }..];
```

Reply:
And please add a test to cover this case :)

Reply:
I think I did the clone/collect thing because I was thinking about the general case of "characters in Rust can be up to four bytes long", but I guess if we know the first character is an ASCII `.`, then we're free to slice it.